common/app/routes/Profile/components/HeatMap.jsx

```diff
@@ -57,7 +57,10 @@ class HeatMap extends Component {
     const rectSelector = '#cal-heatmap > svg > svg.graph-legend > g > rect.r';
     const calLegendTitles = ['0 items', '1 item', '2 items', '3 or more items'];
     const firstTS = Object.keys(calendar)[0];
-    let start = new Date(firstTS * 1000);
+    // start should not be earlier than 7 months before now:
+    let start = (firstTS * 1000 + 1000 * 60 * 60 * 24 * 210 < Date.now())
```

Comment:
It would be better to store the expression result as constant and then use it here.

Reply:
Hi @vikaskyadav ,
just to confirm, do you mean 7 months time span?
Example?
![image](https://user-images.githubusercontent.com/18584913/43126528-a09432f8-8f35-11e8-863e-f9aec6bbcfeb.png)

Reply:
Yes. This is definitely better. Please update the commit to reflect the same.